lib/routes/exposures/index.js

```diff
@@ -18,6 +18,38 @@ const {
 } = require('./query')
 
 async function exposures(server, options, done) {
+  /**
+   * Allows users to validate the verification code they get when given a positive diagnosis. The request contains a
+   * 256 character hash made up of two 128 character hashes. The first 128 characters contain a control code while the
+   * second 128 character hash contains the actual verification code given to the user.
+   *
+   * See schema.js/verify for details oon the input/output structure.
+   *
+   * First, we verify the user is within configured rate limits (same user can't verify more than 1 code per second,
+   * regardless of the code value).
+   *
+   * Next, we verify that the control code is within configured rate limits (the first
+   * 3 digits of the code cannot be verified more than once per second). This means that codes 123456 and 123987
+   * cannot be verified within the same second, regardless of the user verifying either. This check exists to help
+   * limit attacks if someone is registering new user records to bypass user rate limit checks. I don't know how
+   * effective such a check will end up being, and it does add a database write. The result of this database call
+   * is again checked against rate limiting, but I'm not quite sure why the double check is done - perhaps to
+   * protect against slow database connections?
+   *
+   * Next, we delete the verification record, using both the control and full code to delete only a single record.
+   * This call returns when the code record was created and the user's onset date (I assume onset of symptoms). The
+   * code TTL is verified to be within the configured lifetime, using .env value `CODE_LIFETIME_MINS`. On dev this
+   * is set to CODE_LIFETIME_MINS=10.
+   *
+   * And finally, the user's ID and onset date is inserted into the upload_tokens table. This call results in the
+   * generation of a new upload_token ID value, which is returned to the caller.
+   *
+   * Responses:
+   *  200: Upload token, if everything is successful
+   *  403: If the code doesn't exist
```

Comment:
This should be 404 no?

Reply:
The current code implementation returns a 403 - Forbidden in 2 cases:

1. The rate limit check/write on table `verifications` fails either because the control doesn't exist or too many requests within a period of time.
1. The code/control cannot be deleted from the `verifications` table.

Whether a 404 is a better result code is up for debate, but the documented 403 is accurate as currently implemented. That said, the reason could be clarified to include the rate limit check.

Reply:
Added additional clarity in latest commit.